Client/sources/components/fileListViewer/fileListDisplay.tsx

```diff
@@ -1,42 +1,60 @@
 import * as React from "react";
 import { IFile } from "../../interfaces/file.interface";
 import { IFolder } from "../../interfaces/folder.interface";
-import { downloadFile, fetchAllFiles } from "./utils";
+import { downloadFile, fetchAllFiles , fetchDirectoryByPath } from "./utils";
 import { FileList } from "../fileList";
-
+import { GlobalContext } from "../../contexts/globalContext";
 /* 
 Fetches all Files from Server
 Maintains it's own state and shows currentDirectoryFiles
 */
 export const FileViewer = () => {
-  const [filesData, setFilesData] = React.useState<IFolder | undefined>(
-    undefined
-  );
-  const [currentDir, setCurrentDir] = React.useState<IFolder | undefined>(
+  
+
+  const { currentDir , setCurrentDir , filesData , setFilesData } = React.useContext(GlobalContext);
+ /* const [currentDir, setCurrentDir] = React.useState<IFolder | undefined>(
     undefined
-  );
-  useFetchFilesEffect(setFilesData, setCurrentDir);
-  const directories = useMemoizedFileDataPrepare(currentDir);
+  );*/
+  useFetchFilesEffect(setFilesData, setCurrentDir); 
+
+  const directories = useMemoizedFileDataPrepare(currentDir);   //figures out all files/folders within currentDir
+
   const {
     directoryOnClickCallback,
     filesOnClickCallback,
   } = useOnClickCallbacks(setCurrentDir, currentDir);
+
+  //handles the back button click and moves one level up
+  const levelUp = async () =>{
+    const tempPath = currentDir.path.slice(0,currentDir.path.lastIndexOf('/'));
+    if(tempPath === filesData.path)
+      setCurrentDir(filesData);
+    else   
+      setCurrentDir(await fetchDirectoryByPath(tempPath));
+  }
+
   return (
     <>
       <button
-        disabled={currentDir === filesData}
+        disabled={currentDir == filesData}
         onClick={() => {
-          setCurrentDir(filesData);
+        setCurrentDir(filesData);
         }}
```

Comment:
Extract this to a separate function at module level and useCallback to save computation